spec/post_spec.rb

```diff
@@ -33,4 +33,34 @@
       expect(post).to be_valid
     end
   end
+
+  context 'recent_comments' do
+    it 'returns the most recent comments, limited to 5' do
+      user = User.create(name: 'John')
+      post = Post.create(author: user, title: 'My Post', text: 'Some text')
+
+      newest_comment = post.comments.create(user:, text: 'This is the newest comment', created_at: Time.now)
+      middle_comment1 = post.comments.create(user:, text: 'This is a middle comment 1', created_at: 1.day.ago)
+      middle_comment2 = post.comments.create(user:, text: 'This is a middle comment 2', created_at: 2.days.ago)
+      middle_comment3 = post.comments.create(user:, text: 'This is a middle comment 3', created_at: 3.days.ago)
+      older_comment = post.comments.create(user:, text: 'This is an older comment', created_at: 4.days.ago)
+
+      post.comments.create(user:, text: 'This is an extra comment', created_at: 6.days.ago)
+
+      recent_comments = post.recent_comments
+
+      expect(recent_comments).to eq([newest_comment, middle_comment1, middle_comment2, middle_comment3, older_comment])
+    end
+  end
+
+  context 'private methods' do
+    it 'updates the author\'s posts_counter' do
+      user = User.create(name: 'testuser')
+      post = Post.new(author: user, title: 'My Post', text: 'Some text')
+      post.save
+
+      user.reload
+      expect(user.posts_counter).to eq(1)
+    end
+  end
```

Comment:
- [ ] Well done testing the `recent_comments` method and the `recent_posts` method in the `User` model. However, as requested by the previous reviewer please ensure to test the `update_post_counter` method as well. 👌 😊

Reply:
Hi @ridwanediallo is exactly what is doing. This is the code of `post.rb`:

```
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User', foreign_key: :author_id
  has_many :likes, foreign_key: :post_id
  has_many :comments, foreign_key: :post_id
  after_save :update_post_counter

  validates :title, presence: true, length: { maximum: 250 }
  validates :comments_counter, numericality: { greater_than_or_equal_to: 0 }
  validates :likes_counter, numericality: { greater_than_or_equal_to: 0 }

  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

  private

  def update_post_counter
    author.update(posts_counter: author.posts.count)
  end
end
```

As you can see the method `update_post_counter` is updating the `post_counter` attribute associated to the 'User' class, as you can see in the snipet of `user.rb`.

```
class User < ApplicationRecord
  has_many :comments, foreign_key: :user_id
  has_many :likes, foreign_key: :user_id
  has_many :posts, foreign_key: :author_id

  validates :name, presence: true
  validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }

  def recent_posts
    posts.order(created_at: :desc).limit(3)
  end
end
```

Please kindly review it again. Thanks!